daemon/logger/factory/factory.go

```diff
@@ -0,0 +1,57 @@
+package factory
+
+import (
+	"fmt"
+	"sync"
+
+	logger "github.com/docker/docker/daemon/logger"
+)
+
+// Creator is a method that builds a logging driver instance with given context
+type Creator func(LoggingContext) (logger.Logger, error)
+
+// LoggingContext provides enough information for a logging driver to do its function
+type LoggingContext struct {
```

Comment:
@LK4D4 OK, one thing at a time... We need "some" unified way (i.e. type Creator above) to initiate all logging drivers the same way every time a container is created, right?

Passing ResourcePathProvider here is happening b/c the "-json" string lives in the jsonlogdriver pkg and moving that string back to daemon/container.go (which is totally against what this PR is trying to achieve), and generating the logfile path there and passing it back here is sort of sounds worse. In case of drivers that don't need to use this container's resource path, they may choose to not to use this field and they're ok...

Do you have a better solution?